tests/unit/pubsub_v1/subscriber/test_streaming_pull_manager.py

```diff
@@ -144,6 +144,15 @@ def test_ack_deadline():
     assert manager.ack_deadline == 20
 
 
+def test_ack_deadline_w_max_extension_period():
+    manager = make_manager()
+    manager._flow_control = types.FlowControl(max_extension_period=5)
+    for _ in range(5):
+        manager.ack_histogram.add(20)
+
+    assert manager.ack_deadline == 5
```

Comment:
Adding `20` five times in a row to fill histogram with data. With this, `max_extension_period` should be chosen

Reply:
Can you also add 
"assert manager.ack_deadline == 5"
before line 150?

This way we check that the max_extension_period is respected even if there have been no add() calls to ack_histogram.

Reply:
Oh, if `max_extension_period` should be respected from the very beginning, then property can be written little bit easier. Pushed